Helm: ACL token env var is incorrect for Sync Catalog

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request. Searching for pre-existing feature requests helps us consolidate datapoints for identical requirements into a single place, thank you!
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.

<!--- Thank you for keeping this note for the community --->

---

<!--- When filing a bug, please include the following headings if possible. Any example text in this template can be deleted. --->

### Overview of the Issue

When configuring [syncCatalog.aclSyncToken](https://developer.hashicorp.com/consul/docs/k8s/helm#v-synccatalog-aclsynctoken) the helm chart configures `CONSUL_HTTP_TOKEN` but the control plane command is now looking for a token at `CONSUL_ACL_TOKEN`


<!--- Please describe the issue you are having and how you encountered the problem. --->

### Reproduction Steps

Configure `syncCatalog.aclSyncToken` in your Helm values to point at an existing Kubernetes secret containing an ACL token

### Logs

```
2023-02-07T22:43:44.105Z [ERROR] consul-server-connection-manager: connection error: error="fetching supported dataplane features: rpc error: code = PermissionDenied desc = Permission denied: token with AccessorID '00000000-0000-0000-0000-000000000002' lacks permission 'service:write' on \"any service\""
```


### Expected behavior

Sets `CONSUL_ACL_TOKEN` 

### Environment details
Consul-k8s: 1.0.3

External Consul servers
ACLs / Tokens managed externally too


### Additional Context

Upgrading from Consul-k8s 0.49

It seems odd and confusing to have changed from using the [standard](https://developer.hashicorp.com/consul/commands#environment-variables) `CONSUL_HTTP_TOKEN` environment variable that every other Consul tool uses.

--- a/charts/consul/templates/sync-catalog-deployment.yaml
+++ b/charts/consul/templates/sync-catalog-deployment.yaml
@@ -100,7 +100,7 @@ spec:
                 fieldRef:
                   fieldPath: metadata.namespace
             {{- if (and .Values.syncCatalog.aclSyncToken.secretName .Values.syncCatalog.aclSyncToken.secretKey) }}
-            - name: CONSUL_HTTP_TOKEN
+            - name: CONSUL_ACL_TOKEN
               valueFrom:
                 secretKeyRef:
                   name: {{ .Values.syncCatalog.aclSyncToken.secretName }}

--- a/charts/consul/test/unit/sync-catalog-deployment.bats
+++ b/charts/consul/test/unit/sync-catalog-deployment.bats
@@ -338,7 +338,7 @@ load _helpers
       --set 'syncCatalog.enabled=true' \
       --set 'syncCatalog.aclSyncToken.secretKey=bar' \
       . | tee /dev/stderr |
-      yq '[.spec.template.spec.containers[0].env[].name] | any(contains("CONSUL_HTTP_TOKEN"))' | tee /dev/stderr)
+      yq '[.spec.template.spec.containers[0].env[].name] | any(contains("CONSUL_ACL_TOKEN"))' | tee /dev/stderr)
   [ "${actual}" = "false" ]
 }
 
@@ -349,7 +349,7 @@ load _helpers
       --set 'syncCatalog.enabled=true' \
       --set 'syncCatalog.aclSyncToken.secretName=foo' \
       . | tee /dev/stderr |
-      yq '[.spec.template.spec.containers[0].env[].name] | any(contains("CONSUL_HTTP_TOKEN"))' | tee /dev/stderr)
+      yq '[.spec.template.spec.containers[0].env[].name] | any(contains("CONSUL_ACL_TOKEN"))' | tee /dev/stderr)
   [ "${actual}" = "false" ]
 }
 
@@ -361,7 +361,7 @@ load _helpers
       --set 'syncCatalog.aclSyncToken.secretName=foo' \
       --set 'syncCatalog.aclSyncToken.secretKey=bar' \
       . | tee /dev/stderr |
-      yq '[.spec.template.spec.containers[0].env[].name] | any(contains("CONSUL_HTTP_TOKEN"))' | tee /dev/stderr)
+      yq '[.spec.template.spec.containers[0].env[].name] | any(contains("CONSUL_ACL_TOKEN"))' | tee /dev/stderr)
   [ "${actual}" = "true" ]
 }
 

